tests/numeric/test_PCATransformer.py

```diff
@@ -0,0 +1,416 @@
+            match=r"""PCATransformer:n_components must be positive  got -1""",
+        ):
+            PCATransformer(columns="b", n_components=-1)
+
+    def test_super_init_called(self, mocker):
+        """Test that super.__init__ called."""
+
+        expected_call_args = {
+            0: {
+                "args": (),
+                "kwargs": {"columns": ["a", "b"], "copy": True, "verbose": False},
+            }
+        }
+
+        with ta.functions.assert_function_call(
+            mocker, tubular.base.BaseTransformer, "__init__", expected_call_args
+        ):
+            PCATransformer(columns=["a", "b"], n_components=1, copy=True, verbose=False)
+
+
+class TestCheckNumericColumns(object):
+    """Tests for the check_numeric_columns method."""
+
+    def test_arguments(self):
+        """Test that check_numeric_columns has expected arguments."""
+
+        ta.functions.test_function_arguments(
+            func=PCATransformer.check_numeric_columns,
+            expected_arguments=["self", "X"],
+            expected_default_values=None,
+        )
+
+    def test_exception_raised(self):
+        """Test an exception is raised if non numeric columns are passed in X."""
+
+        df = d.create_df_2()
+
+        x = PCATransformer(columns=["a", "b", "c"], n_components=2)
+
+        with pytest.raises(
+            TypeError,
+            match=r"""PCATransformer: The following columns are not numeric in X; \['b', 'c'\]""",
+        ):
+            x.check_numeric_columns(df)
+
+    def test_X_returned(self):
+        """Test that the input X is returned from the method."""
+
+        df = d.create_df_2()
+
+        x = PCATransformer(columns=["a"], n_components=2)
+
+        df_returned = x.check_numeric_columns(df)
+
+        ta.equality.assert_equal_dispatch(
+            expected=df,
+            actual=df_returned,
+            msg="unexepcted object returned from check_numeric_columns",
+        )
+
+
+class TestFit(object):
+    """Tests for PCATransformer.fit()."""
+
+    def test_arguments(self):
+        """Test that fit has expected arguments."""
+
+        ta.functions.test_function_arguments(
+            func=PCATransformer.fit,
+            expected_arguments=["self", "X", "y"],
+            expected_default_values=(None,),
+        )
+
+    def test_super_fit_call(self, mocker):
+        """Test the call to BaseTransformer.fit."""
+
+        df = created_numeric_df_1()
+
+        x = PCATransformer(columns=["a", "b"], n_component=1)
+
+        expected_call_args = {0: {"args": (created_numeric_df_1(), None), "kwargs": {}}}
+
+        with ta.functions.assert_function_call(
+            mocker, tubular.base.BaseTransformer, "fit", expected_call_args
+        ):
+            x.fit(df)
+
+    def test_check_numeric_columns_call(self, mocker):
+        """Test the call to PCATransformer.check_numeric_columns."""
+
+        df = created_numeric_df_1()
+
+        x = PCATransformer(columns=["a", "b"], n_component=1)
```

Comment:
n_components

Reply:
Commit : [fix typo n_components in test](https://github.com/lvgig/tubular/pull/57/commits/b740a66eb635da9295658ce7341941f2e86a07a6)